### PR TITLE
Remove external Brave-managed autoplay whitelist

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/8408

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by pes10k** [comment]
re: https://github.com/brave/brave-browser/issues/8408

**Comment by pilgrim-brave** [comment]
> shouldn't we also register the default value to be `CONTENT_SETTING_ALLOW` in `chromium_src/components/content_settings/core/browser/content_settings_registry.cc`?

Why? Is this a product change?

**Comment by darkdh** [comment]
>> shouldn't we also register the default value to be `CONTENT_SETTING_ALLOW` in `chromium_src/components/content_settings/core/browser/content_settings_registry.cc`?

>Why? Is this a product change?

The autoplay list was to mitigate the web compatibility issue when we have default block autoplay global setting.  According to https://github.com/brave/brave-browser/issues/8408, we want to run chromium's policy first and then respect user's content setting. So with block by default content setting, even chromium allows auotplay, media still cannot be autoplayed unless user explicitly allow it in per site content setting. That is why we need to set it to allow by default and user can still have control if they need to block autoplay per site.

**Comment by bsclifton** [comment]
@pilgrim-brave does this fix https://github.com/brave/brave-browser/issues/8408?

If so, can you update the original post 😄 Thanks!

**Comment by pes10k** [comment]
Also, are there CRX changes that should be made after this is on all platforms?  Just to make sure we're no longer bundling the list?  If so, and there isn't already an issue, i'll create one

**Comment by pilgrim-brave** [comment]
Hey all, I lost a commit (my fault) so don't review this yet.

**Comment by pilgrim-brave** [comment]
> According to [brave/brave-browser#8408](https://github.com/brave/brave-browser/issues/8408), we want to run chromium's policy first and then respect user's content setting. So with block by default content setting, even chromium allows auotplay, media still cannot be autoplayed unless user explicitly allow it in per site content setting. That is why we need to set it to allow by default and user can still have control if they need to block autoplay per site.

OK this set off a cascade of changes which required some rewriting of the logic in our `AllowAutoplay`, but I think I have it now.

**Comment by pilgrim-brave** [comment]
As per the linked mini-spec, I've leaving all of the Brave-specific per-site autoplay content settings intact. In the absence of a per-site or rule-based autoplay content setting from the user (which will be most sites), we now do what upstream does.

**Comment by pilgrim-brave** [comment]
Ready for (re-)review.

File: patches/chrome-VERSION.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 diff --git a/chrome/VERSION b/chrome/VERSION
-index 02faec526acf56c05150c1a87b90e3b57ddefd87..19faa8c75a86e3b39ef606ad0f62f2d774d005d7 100644
+index 02faec526acf56c05150c1a87b90e3b57ddefd87..5a1f13e4275aca35d3dc214d57e81276027ce02b 100644
```
Comment (darkdh) [inline patches/chrome-VERSION.patch]: I believe this was added by accident

**Comment by pilgrim-brave** [comment]
> I think we should also remove `BRAVE_AUTOPLAY_POLICY_DOCUMENT_SHOULD_AUTOPLAY_MUTED_VIDEOS` patches and overrides to match upstream policy
> otherwise overall looks good to me.

Done

**Comment by pilgrim-brave** [comment]
AFAICT this is working and passes all tests, but we still have not had a successful Jenkins run because

- lint fails with an unrelated error (`'Changelist' object has no attribute 'GetChange'`)
- audit deps fails with a number of unrelated errors
- tests do not run because of an unrelated speedreader error

**Pull request merged**